clippy_lints/Cargo.toml

```diff
@@ -19,8 +19,8 @@ pulldown-cmark = { version = "0.9", default-features = false }
 quine-mc_cluskey = "0.2"
 regex-syntax = "0.6"
 serde = { version = "1.0", features = ["derive"] }
-serde_json = { version = "1.0", optional = true }
-tempfile = { version = "3.3.0", optional = true }
+serde_json = "1.0"
+tempfile = "3.3.0"
```

Comment:
Those deps are pretty big. I'm not sure about compiling them into Clippy. Can you check the final binary sizes and how they change with those deps? Also compile times?

Reply:
On this branch:
Compile-time:
```
$ cargo clean
$ cargo build -r
```
58.61s

Sizes: 
`target/release/clippy-driver`: 12M
`target/release/cargo-clippy`: 1.3M

---

On `master`:
Times:
```
$ cargo clean
$ cargo build -r
```
52.08s

Sizes:
`target/release/clippy-driver`: 11M
`target/release/cargo-clippy`: 796K

---

You're right, they're pretty heavy. I'll see if I can make them optional in some way.